src/managed/Microsoft.NET.HostModel/AppHost/MachOUtils.cs

```diff
@@ -0,0 +1,302 @@
+                                    symtab = (SymtabCommand*)command;
+                                }
+                                else if (command->cmd == Command.LC_CODE_SIGNATURE)
+                                {
+                                    Verify(i == header->ncmds - 1, MachOFormatError.SignCommandNotLast);
+                                    signature = (LinkEditDataCommand*)command;
+                                    break;
+                                }
+
+                                file += command->cmdsize;
+                            }
+
+                            if (signature != null)
+                            {
+                                Verify(linkEdit != null, MachOFormatError.SignNeedsLinkEdit);
+                                Verify(symtab != null, MachOFormatError.SignNeedsSymtab);
+
+                                var symtabEnd = symtab->stroff + symtab->strsize;
+                                var linkEditEnd = linkEdit->fileoff + linkEdit->filesize;
+                                var signatureEnd = signature->dataoff + signature->datasize;
+                                var fileEnd = (ulong)stream.Length;
+
+                                Verify(linkEditEnd == fileEnd, MachOFormatError.LinkEditNotLast);
+                                Verify(signatureEnd == fileEnd, MachOFormatError.SignBlobNotLast);
+
+                                Verify(symtab->symoff > linkEdit->fileoff, MachOFormatError.SymtabNotInLinkEdit);
+                                Verify(signature->dataoff > linkEdit->fileoff, MachOFormatError.SignNotInLinkEdit);
+
+                                // The signature blob immediately follows the symtab blob, 
+                                // except for a few bytes of padding.
+                                Verify(signature->dataoff >= symtabEnd && signature->dataoff - symtabEnd < 32, MachOFormatError.SignBlobNotLast);
+
+                                // Remove the signature command
+                                header->ncmds--;
+                                header->sizeofcmds -= signature->cmdsize;
+                                Unsafe.InitBlock(signature, 0, signature->cmdsize);
+
+                                // Remove the signature blob (note for truncation)
+                                signatureSize = (uint)(fileEnd - symtabEnd);
+
+                                // Adjust the __LINKEDIT segment load command
+                                linkEdit->filesize -= signatureSize;
+
+                                // codesign --remove-signature doesn't reset the vmsize.
+                                // Setting the vmsize here makes the output bin-equal with the original 
+                                // unsigned apphost (and not bin-equal with a signed-unsigned-apphost).
+                                linkEdit->vmsize = linkEdit->filesize;
```

Comment:
I wonder if this confuses tools such as `size` or whatever is the equivalent of `readelf` for Mach-O.

Reply:
* I've verified that `size` and `otool` work fine on the generated apphost.
* The generated host is bit-equivalent to the original apphost (before it was signed at all).

So, I think the updated are OK.